stl/inc/format

```diff
@@ -2650,6 +2658,48 @@ struct _Arg_formatter {
     }
 };
 
+
+// Special compile time version of _Parse_format_specs. This version is parameterized on
+// the type of the argument associated with the format specifier, since we don't really
+// care about avoiding code bloat for code that never runs at runtime, and we can't form
+// the erased basic_format_args structure at compile time.
+template <typename _Ty, typename _ParseContext>
+constexpr typename _ParseContext::iterator _Compile_time_parse_format_specs(_ParseContext& _Pc) {
+    using _CharT                = typename _ParseContext::char_type;
+    using _Context              = basic_format_context<_Fmt_it, _CharT>;
+    using _ArgTraits            = _Format_arg_traits<_Context>;
+    using _FormattedTypeMapping = typename _ArgTraits::template _Storage_type<_Ty>;
+    // If the type is going to use a custom formatter we should just use that,
+    // instead of trying to instantiate a custom formatter for it's erased handle
+    // type
+    using _FormattedType = conditional_t<is_same_v<_FormattedTypeMapping, typename basic_format_arg<_Context>::handle>,
+        _Ty, _FormattedTypeMapping>;
+    auto _Formatter      = formatter<_FormattedType, _CharT>();

```

Comment:
I believe that can simply be
```suggestion
    formatter<_FormattedType, _CharT> _Formatter;
```

Maybe we could even go with this, although I am not really a fan
```suggestion
    return formatter<_FormattedType, _CharT>{}._Parse(_Pc);
```

Reply:
No, these are required to be written this way.

I should add a note indicating such with a pointer to the bug number.

(the {}s cause cl to fail to parse it as a type, because FormattedType is dependent on a dependent template, it might work now that the template keyword isn't needed specifically in the declaration of `_Formatter`.